streaming/base/shared/prefix.py

```diff
@@ -43,61 +43,64 @@ def _get_path(prefix_int: int, name: str) -> str:
     return f'{prefix_int:06}_{name}'
 
 
-def _pack_locals(dirnames: Set[str]) -> bytes:
-    """Pack local dirnames.
+def _pack_locals(dirnames: Set[str], prefix_int: int) -> bytes:
+    """Pack local dirnames and prefix int.
 
     Args:
         dirnames (Set[str]): Unpacked local dirnames.
+        prefix_int (int): Prefix int.
 
     Returns:
-        bytes: Packed local dirnames.
+        bytes: Packed local dirnames and prefix int.
     """
-    text = '\0'.join(sorted(dirnames))
+    text = '\0'.join(sorted(dirnames)) + f'\0{prefix_int}'
     data = text.encode('utf-8')
     size = 4 + len(data)
     return b''.join([np.int32(size).tobytes(), data])
 
 
-def _unpack_locals(data: bytes) -> Set[str]:
-    """Unpack local dirnames.
+def _unpack_locals(data: bytes) -> Tuple[Set[str], int]:
+    """Unpack local dirnames and prefix int.
 
     Args:
-        data (bytes): Packed local dirnames.
+        data (bytes): Packed local dirnames and prefix int.
 
     Returns:
-        Set[str]: Unpacked local dirnames.
+        Set[str]: Unpacked local dirnames and prefix int.
     """
     size = np.frombuffer(data[:4], np.int32)[0]
     text = data[4:size].decode('utf-8')
-    return set(text.split('\0'))
+    text = text.split('\0')
+    return set(text[:-1]), int(text[-1] or 0)
 
 
-def _check_self(my_locals: List[str]) -> Set[str]:
+def _check_self(streams_local: List[str]) -> Set[str]:
```

Comment:
just wondering, any reason for the name change? not asking for an edit but jw

Reply:
Changed just for better understanding